src/openforms/submissions/form_logic.py

```diff
@@ -147,21 +177,23 @@ def evaluate_form_logic(
                     updated_step_data = {}
                     step._is_applicable = False
             elif action_details["type"] == LogicActionTypes.variable:
-                new_value = jsonLogic(action_details["value"], updated_submission_data)
+                new_value = jsonLogic(action_details["value"], data_container.data)
                 variable_key = action["variable"]
                 updated_step_data[variable_key] = new_value
-                updated_submission_data = {**data, **updated_step_data}
+                data_container.update_data(updated_step_data)
     step.data = DirtyData(updated_step_data)
 
     # Logging the rules
-    logevent.submission_logic_evaluated(submission, evaluated_rules, data)
+    logevent.submission_logic_evaluated(
+        submission, evaluated_rules, data_container.data
```

Comment:
this should be `{**data_container.static_data, **data_container.initial_data}` - now you're still passing the mutated data to the log event instead of the original input data.